scantool/scantool.c

```diff
@@ -640,6 +640,15 @@ l2_do_send(struct diag_l2_conn *d_conn, void *data, size_t len, void *handle)
 
 	msg.len = len;
 	msg.data = (uint8_t *)data;
+
+	if (d_conn->l2proto->diag_l2_protocol == DIAG_L2_PROT_VAG) {
```

Comment:
The l2_vag code currently uses msg.type for the block title (command opcode) in contrast to the other L2s that use the first byte of the message.

l2_do_send is only called by the 'diag sendreq' and '850 sendreq' commands.